BinaryBenchmark.Bebop.cs

```diff
@@ -18,9 +19,20 @@ public User[] Bebop_Deserialize(BebopDataSet data)
         return Wrapper.Decode(data.Data).Users;
     }
     
-    private byte[] DataConvert_Bebop(User[] users)
+    private ReadOnlyMemory<byte> DataConvert_Bebop(User[] users)
     {
-        return Wrapper.Encode(new Wrapper { Users = users });
+        var wrapper = new Wrapper { Users = users };
+        var size = wrapper.MaxByteCount;
+        var encodeBuffer = ArrayPool<byte>.Shared.Rent(size);
+        try
+        {
+            var written = wrapper.EncodeIntoBuffer(encodeBuffer);
+            return new ReadOnlyMemory<byte>(encodeBuffer, 0, written);
```

Comment:
This ReadOnlyMemory will reference a buffer which has been returned back to the array pool and can therefore be rented by another thread and modified while it is still being held. This is very dangerous and will break in a real application.

Reply:
> This ReadOnlyMemory will reference a buffer which has been returned back to the array pool and can therefore be rented by another thread and modified while it is still being held. This is very dangerous and will break in a real application.

Thank you for pointing this out for any potential third-parties. Correct, you should never do this in a real application; in a real app you’d use the buffer (send it over the wire) and free it.